src/Orleans.Runtime/GrainDirectory/MultiClusterRegistration/GlobalSingleInstanceActivationMaintainer.cs

```diff
@@ -95,10 +101,21 @@ public void TrackDoubtfulGrains(Dictionary<GrainId, IGrainInfo> newstuff)
             {
                 try
                 {
```

Comment:
The `AsynchAgent.Run()` method should not be `async void`, it's supposed to be synchronous and run in its own thread.

This will manifest as:
  * Tasks scheduled here will be continued on `TaskScheduler.Default`, not the agent's thread.
  * The logs will print out a message like "Starting AsyncAgent X on managed thread Y" immediately followed by "Stopping AsyncAgent X that runs on managed thread Y".
  * The agent's fault handling policies don't get a chance to take effect.

That behavior is all probably fine here, it seems to defeat the purpose of using `AsynchAgent`.

Since this isn't new to this PR, we don't need to change it here, but it's worth noting.

Reply:
Makes sense. May as well change it in this PR.